pg_auth_mon.c

```diff
@@ -214,6 +266,8 @@ auth_monitor(Port *port, int status)
 		fai->last_successful_login_at = GetCurrentTimestamp();
 	}
 
+	log_pg_auth_mon_data();
+
```

Comment:
In my opinion, instead of keeping the exclusive lock for all this log duration, I would suggest to call this after LWLockRelease and then in the function itself take shared lock. As we are only reading from the hash table, and a little out of date entries in logs is more tolerable than delays in log in attempts to the db.

Reply:
I decided no to do exactly that to avoid the complexity of working with extra locks because here potential performance gains are unlikely to justify that. The current implementation is correct in a sense it ensures mutual exclusion for the hash table access; any performance benefits of a finer-grained locking are likely to be visible only with extreme numbers of connections that we very rarely see.

Reply:
Actually even that fact that we have run `pg_aut_mon` with the global exclusive lock to manage the access to the hash table (and not some "shared lock on the table - lookup - exclusive lock on the entry" locking scheme ) without any problems for quite some time now indicates such performance optimizations are not strictly necessary.